src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private static final String INDEX_OUT_OF_BOUNDS_EXCEPTION_MESSAGE
+            = "Invalid index value! Actual list size is: ";
+    private static final String NO_SUCH_ELEMENT_EXCEPTION_MESSAGE
+            = "Element not found! Invalid value!";
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = createListContainer(INITIAL_CAPACITY);
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            indexValidation(index);
+            shiftRight(index);
+            elementData[index] = value;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexValidation(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        T deletedElement = elementData[index];
+        shiftLeft(index);
```

Comment:
The same, don't complicate your logic
```suggestion
         System.arraycopy(elementData, index + 1, elementData, index, --size - index);
```